app/scripts/modules/core/src/application/nav/AppRefresher.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Observable, Subject } from 'rxjs';
+
+import { Application } from '../application.model';
+import { ApplicationDataSource, IFetchStatus } from '../service/applicationDataSource';
+
+import { AppRefresherIcon } from './AppRefresherIcon';
+
+export interface IAppRefresherProps {
+  app: Application;
+}
+
+export const AppRefresher = ({ app }: IAppRefresherProps) => {
+  const app$ = new Subject<Application>();
+  const destroy$ = new Subject();
```

Comment:
This won't work the way you want because a new `app$` and `destroy$` is created for every render.

Reply:
My instinct is to remove most of the rxjs and just use hooks.  I think it would simplify the code.